great_expectations/experimental/datasources/filesystem_data_asset.py

```diff
@@ -53,6 +153,77 @@ def build_batch_request(
 
         return super().build_batch_request(options)
 
+    def get_batch_list_from_batch_request(
+        self, batch_request: BatchRequest
+    ) -> list[Batch]:
+        self._validate_batch_request(batch_request)
+        batch_list: list[Batch] = []
+
+        kwargs: dict | None = getattr(self, "kwargs", None)
+        if not kwargs:
+            kwargs = {}
+
+        for request, path in self._fully_specified_batch_requests_with_path(
+            batch_request
+        ):
+            batch_spec = PathBatchSpec(
+                path=str(path),
+                reader_method=self._get_reader_method(),
+                reader_options=self.dict(
+                    include=self._get_reader_options_include(),
+                    exclude=self._EXCLUDE_FROM_READER_OPTIONS,
+                    exclude_unset=True,
+                    by_alias=True,
+                    **kwargs,
+                ),
+            )
+            execution_engine: PandasExecutionEngine | SparkDFExecutionEngine = (
+                self.datasource.get_execution_engine()
+            )
+            data, markers = execution_engine.get_batch_data_and_markers(
+                batch_spec=batch_spec
+            )
+
+            # batch_definition (along with batch_spec and markers) is only here to satisfy a
+            # legacy constraint when computing usage statistics in a validator. We hope to remove
+            # it in the future.
+            # imports are done inline to prevent a circular dependency with core/batch.py
+            from great_expectations.core import IDDict
+            from great_expectations.core.batch import BatchDefinition
+
+            batch_definition = BatchDefinition(
+                datasource_name=self.datasource.name,
+                data_connector_name="experimental",
+                data_asset_name=self.name,
+                batch_identifiers=IDDict(request.options),
+                batch_spec_passthrough=None,
+            )
+
+            batch_metadata = copy.deepcopy(request.options)
+            batch_metadata["base_directory"] = self.datasource.base_directory
```

Comment:
@NathanFarmer Instead of `base_directory`, we can use `PathBatchSpec` one level above (and thus remove this module).  This is proposed in #7197.